### ISSUE TITLE
Add Support for a Custom Test Type

### ISSUE THREAD

**Issue by VincenzoFerme** [benchflow/benchflow#233]
_From @VincenzoFerme on August 22, 2016 14:47_

@simonedavico document here the work that has to be done, and where, to implement the simplest (and the base for the other) test, that is the `Custom Test`. First describe what a `Custom Test` is, then elaborate on the work that has to be done to implement it (of course, at an high level of details)

As part of the Custom Test Type, we might want to offer some test types specific to the WfMSs, as for example exploration that not only applies to the workload or the resources and SUT configurations, but also to the structure and characteristic of the workload mix. @ivanchikj describe the requirements (not the why) of those test referring to the terminology used in the Simone's thesis.

_Copied from original issue: benchflow/test-orchestrator#5_

**Comment by jeppe-style** [comment]
Is this still up-to-date?

**Comment by VincenzoFerme** [comment]
Yes, lets keep it open. This is the most general test type, that is similar to what we call now `configuration` test.
